twenty-first/src/util_types/mmr/mmr_membership_proof.rs

```diff
@@ -221,12 +229,32 @@ impl<H: AlgebraicHasher> MmrMembershipProof<H> {
     /// Batch update multiple membership proofs.
     /// Returns the indices of the membership proofs that were modified where index refers
     /// to the order in which the membership proofs were given to this function.
+    /// Panics if `membership_proofs` and `membership_proof_leaf_indices` do not have
+    /// the same length, or if a leaf index is out-of-bounds for the MMR.
     pub fn batch_update_from_append(
         membership_proofs: &mut [&mut Self],
+        membership_proof_leaf_indices: &[u64],
```

Comment:
Why not just `leaf_indices`?

Reply:
I like the descriptive name here. I prefer specifying that the leaf-indices relate to the membership proofs and not, e.g., to the indices of the leafs that are appended.